lib/http_server.ex

```diff
@@ -1,18 +1,33 @@
 defmodule HTTPServer do
-  @moduledoc """
-  Documentation for `HTTPServer`.
-  """
+  require Logger
 
-  @doc """
-  Hello world.
+  def accept(port) do
+    tcp_options = [:binary, {:packet, 0}, {:active, false}, reuseaddr: true]
+    {:ok, socket} = :gen_tcp.listen(port, tcp_options)
+    Logger.info("Accepting connections on port #{port}")
+    listen(socket)
+  end
+
+  defp listen(socket) do
+    {:ok, client} = :gen_tcp.accept(socket)
+    spawn(fn -> serve(client) end)
+    listen(socket)
+  end
 
-  ## Examples
+  defp serve(socket) do
+    socket
+    |> read_line()
+    |> write_line(socket)
```

Comment:
My understanding of the [pipe operator](https://elixirschool.com/en/lessons/basics/pipe_operator) is that it passes the result as the first parameter. 

In this case, it seems that you are starting with `socket` and then passing that to `read_line`, then the result of that to `write_line`. But you are also passing `socket` explicitly to `write_line`.  

`write_line` takes two arguments (`data, socket`), so are you passing `socket` as the second argument to `write_line`?  And the first argument that gets passed through the pipe is representative of `data`?

Reply:
> so are you passing socket as the second argument to write_line? And the first argument that gets passed through the pipe is representative of data?

Yes, that's the thought here! I think at one point I had it written out more explicitly:
```
defp serve(socket) do
    data = read_line(socket)
    write_line(data, socket)

    serve(socket)
 end
```
But the version I ended up with was more "elixer-y", for lack of a better word

Reply:
Okay cool I get it now!  That makes sense - thanks!

Reply:
There is a better way; set up ALL the functions so that socket is the first parameter. Then it is more explicit what is going on here.
```suggestion
    socket
    |> read_line()
    |> write_line(daya)
```

Reply:
If you look at the pipe vertically, its as if socket is "moving through the pipe". This is the pattern you will want to follow.

Reply:
I was thinking about this, and it introduces a bit of a puzzle as to how to capture the data. I think this will be solved once you figure out encodings, as your patterns will have to change. Up to you if you want to work on this now or later.

Reply:
I've approved the PR in the meantime